apps/game_backend/lib/game_backend/items.ex

```diff
@@ -244,4 +244,35 @@ defmodule GameBackend.Items do
       {:error, :character_cannot_equip}
     end
   end
+
+  @doc """
+  Gets Item's purchase cost by given currency and item_template.
+  Returns {:ok, purchase_cost} if found one.
+  Returns {:error, :not_found} if there are none.
+  Fails if there are more than one.
+  """
+  def get_purchase_cost_by_currency(currency_id, item_template) do
+    purchase_cost =
+      Map.get(item_template, :purchase_costs)
+      |> Enum.find(fn purchase_cost -> purchase_cost.currency_id == currency_id end)
+
+    case purchase_cost do
+      nil -> {:error, :not_found}
+      purchase_cost -> {:ok, purchase_cost}
+    end
+  end
+
+  @doc """
+  Receives a user_id, an item_template_id and a list of CurrencyCosts.
+  Inserts new Item from given ItemTemplate for given User.
+  Substract the amount of Currency to User by given params.
+  Returns {:ok, map_of_ran_operations} in case of success.
+  Returns {:error, failed_operation, failed_value, changes_so_far} if one of the operations fail.
+  """
+  def buy_item(user_id, template_id, purchase_costs_list) do
+    Multi.new()
+    |> Multi.run(:item, fn _, _ -> insert_item(%{user_id: user_id, template_id: template_id}) end)
+    |> Multi.run(:currencies, fn _, _ -> Currencies.substract_currencies(user_id, purchase_costs_list) end)
```

Comment:
```suggestion
    |> Multi.run(:substract_currencies, fn _, _ -> Currencies.substract_currencies(user_id, purchase_costs_list) end)
```

Reply:
Same as item! This will return a list of currencies if succeeded and I prefer to store them in :currencies key.

Reply:
same as above

Reply:
Thanks!